pymc/deprecator.py

```diff
@@ -0,0 +1,97 @@
+import functools
+import inspect
+import warnings
+import regex as re
+import textwrap
+
+
+def deprecator(reason=None, version=None,action='deprecate', deprecated_args=None, docs=True):
+    """
+    PyMC deprecation helper. This decorator emits deprecated warnings.
+    """    
+    if deprecated_args is not None:
+        deprecated_args = set(deprecated_args.split())
+    cause = reason
+    if cause is not None and version is not None:
+        reason = ", since version " + version + " " + "("+cause+")"
+
+    if cause is not None and version is None:
+        reason = "("+ str(cause) + ")"
+
+    if cause is None and version is not None :
+        reason = ", since version " + version
+    #this function is an edited version of the source code taken from the library Deprecated
+    #https://github.com/tantale/deprecated (MIT License)
+    def sphinxformatter(version, reason):
+        fmtsphinx = ".. deprecated:: {version}" if version else ".. deprecated::"
+        sphinxtext = [fmtsphinx.format(directive = "deprecated", version=version)]
+        width = 2 ** 16
+        reason = reason.lstrip()
+        for paragraph in reason.splitlines():
+            if paragraph:
+                sphinxtext.extend(
+                    textwrap.fill(
+                        paragraph,
+                        width=width,
+                        initial_indent="   ",
+                        subsequent_indent="   ",
+                    ).splitlines()
+                )
+            else:
+                sphinxtext.append("")
+        return sphinxtext
+
+
+    def decorator(func):
+        if deprecated_args is None:
+            if inspect.isclass(func):
+                fmt = "Class {name} is deprecated{reason}."
+            else:
+                fmt = "Function or method {name} is deprecated{reason}."
+        else:
+            fmt = "Parameter(s) {name} deprecated{reason}"
+            
+        if docs==True:
```

Comment:
`if docs is True`

Im not so picky about this but some style guides are very picky. Ill leave it up to you if you want to change it

https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html